node/agent.go

```diff
@@ -825,3 +872,44 @@ func waitForSsNodes(processing map[string]chan int, ssResultChan chan stateResul
 	glog.V(0).Info("All service state nodes are shut down")
 	return
 }
+
+func (a *HostAgent) createContainer(conf *dockerclient.Config, hostConf *dockerclient.HostConfig, svcID string, instanceID int) (*docker.Container, error) {
+	logger := plog.WithFields(log.Fields{
+		"serviceid":  svcID,
+		"instanceid": instanceID,
+	})
+
+	if hostConf == nil {
+		logger.Error("Host Config passed to createContainer is nil.")
+	}
+
+	// create the container
+	opts := dockerclient.CreateContainerOptions{
+		Name:       fmt.Sprintf("%s-%d", svcID, instanceID),
+		Config:     conf,
+		HostConfig: hostConf,
+	}
+
+	if opts.HostConfig == nil {
+		logger.Error("Host Config in opts is nil.")
+	}
+
+	ctr, err := docker.NewContainer(&opts, false, 10*time.Second, nil, nil)
+	if err != nil {
+		logger.WithError(err).Error("Could not create container")
+		return nil, err
+	}
+	logger.WithField("containerid", ctr.ID).Debug("Created a new container")
+	return ctr, nil
+}
+
+func addBindingToMap(bindsMap *map[string]string, cp, rp string) {
```

Comment:
We could declare the type, but I think we'd still need the function - I was pulling out a multi-line repeated code block and making it a one-liner. If we made it a type, we'd still need a function to do the trimming, checking, and logging that was in the original code block. Were you suggesting that we make the function a method of the bindsMap type?

I've changed the param to a regular map (eliminated the pointer), and moved the function into container.go, since it belongs with createContainerConfig.